ClientApp/src/components/ListingIdPage.js

```diff
@@ -0,0 +1,43 @@
+import React, { useEffect, useState } from 'react';
+import { useParams } from 'react-router-dom';
+import PageNotFound from './PageNotFound';
```

Comment:
Currently, PageNotFound is not thrown, in case the id is invalid.
I would suggest in ListingController get by id method change return type to ActionResult and if listing is not found a.k.a null you could return NoContent() or NotFound() and in the other case Ok(listing). And in the frontend, before parsing data check if response.ok and then redirect to pageNotFound or load pageNotFound component something like that.